R/load_files.R

```diff
@@ -1147,6 +1151,97 @@ read_in_results_Tecan <- function(results_file, results_sheets, headers) {
   all_results
 }
 
+#' Load incucyte results from plain text
+#'
+#' This functions loads incucyte time-course cell count file
+#'
+#' @param results_file list of strings: file paths to result paths from individual plates
+#' @param headers list of headers identified in the manifest
+#' @keywords load_files
+#'
+#' @return data.table derived from Incucyte data
+#'
+load_results_Incucyte <-
+  function(results_file,
+           headers = gDRutils::get_env_identifiers()) {
+    
+    # identifiers
+    bcode_name <- headers$barcode[1]
+    d_name <- headers$duration
+    
+    # use a list to store DTs for efficient binding
+    all_data_list <- list()
+    
+    for (iP in results_file) {
+      header_dt <- if (grepl(".xlsx$", iP)) {
+        tryCatch({
+          df <- read_excel_to_dt(iP, n_max = 20)
+        }, error = function(e) {
+          exception_data <- get_exception_data(22)
+          stop(sprintf(exception_data$sprintf_text, iP))
+        })
+      } else {
+        tryCatch({
+          data.table::fread(iP, nrows = 20)
+        }, error = function(e) {
+          exception_data <- get_exception_data(21)
+          stop(sprintf(exception_data$sprintf_text, iP))
+        })
+      }
+      
+      dstart_idx <- which(header_dt[, 1] == "Date Time")
+      if (length(dstart_idx) == 0) { 
+        exception_data <- get_exception_data(37)
+        stop(sprintf(exception_data$sprintf_text, "missing 'Data Time' column"))
+      }
+      barcode_idx <- which(header_dt[, 1] == bcode_name)
+      if (length(barcode_idx) == 0) { 
+        exception_data <- get_exception_data(37)
+        stop(sprintf(exception_data$sprintf_text, "missing 'Barcode' column"))
```

Comment:
Since 
```
 bcode_name <- headers$barcode[1]
```

to not hardcode values:
```suggestion
        stop(sprintf(exception_data$sprintf_text, sprintf("missing '%s' column", bcode_name)))
```